src/map/navi.cpp

```diff
@@ -118,9 +87,15 @@ static int add_path(struct node_heap *heap, int16 x, int16 y, int g_cost, struct
 			tp[i].parent = parent;
 			tp[i].f_cost = g_cost + h_cost;
 			if (tp[i].flag == SET_CLOSED) {
-				heap_push_node(heap, &tp[i]); // Put it in open set again
+				heap.push_back(&tp[i]);
+				std::push_heap(heap.begin(), heap.end(), [](const path_node* a, const path_node* b) {
+					return a->f_cost > b->f_cost;
```

Comment:
in order to prevent rewriting the same comparator, we can create a std::function and use it everywhere
somehting like
```c++
auto heap_comp = [](const path_node* a, const path_node* b) {
    return a->f_cost > b->f_cost;
};
```

and then `std::push_heap(heap.begin(), heap.end(), heap_comp)`